paymentsheet/src/main/java/com/stripe/android/paymentsheet/ConfirmCallback.kt

```diff
@@ -0,0 +1,55 @@
+package com.stripe.android.paymentsheet
+
+import androidx.annotation.RestrictTo
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun interface ConfirmCallback {
+
+    /**
+     * 🚧 Under construction 🚧
+     * The callback to implement to retrieve the intent client secret in the deferred flow.
+     *
+     * @param paymentMethodId the payment method ID to create the intent with
+     *
+     * @return a [ConfirmResponse] which contains the intent client secret and whether or not the
+     * intent was confirmed on the integrators server.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    suspend fun onRetrieveConfirmResponse(paymentMethodId: String): ConfirmResponse
+}
+
+internal fun interface LegacyConfirmCallback : ConfirmCallback {
+    override suspend fun onRetrieveConfirmResponse(paymentMethodId: String): ConfirmResponse {
+        return suspendCoroutine { continuation ->
+            onRetrieveConfirmResponse(
+                paymentMethodId,
+                { continuation.resume(it) },
+                { continuation.resumeWithException(it) }
+            )
+        }
+    }
+
+    fun onRetrieveConfirmResponse(
+        paymentMethodId: String,
+        onSuccess: (ConfirmResponse) -> Unit,
+        onError: (Throwable) -> Unit
```

Comment:
This should mimic the non-legacy callback, so it should include an `onResult: (String) -> ConfirmCallback.Result`.